beetsplug/audible.py

```diff
@@ -103,6 +108,17 @@ def __init__(self):
         )
         self.add_media_field("subtitle", subtitle)
 
+        album_url = mediafile.MediaField(
+            mediafile.MP3StorageStyle("WOAF"),
+            mediafile.MP4StorageStyle("----:com.apple.iTunes:WWWAUDIOFILE"),
+            mediafile.StorageStyle("WOAF"),
+            mediafile.ASFStorageStyle("WM/AudioFileURL"),
+        )
+        self.add_media_field("album_url", album_url)
+
+        region = mediafile.MediaField()
+        self.add_media_field("region", region)
+
```

Comment:
Since the region setting only accepts a list of known good values, would it be possible to validate that the region code is correct? I'm not sure if other beets plugins have this functionality to check configuration correctness.

Reply:
Yes, as [Beets](https://beets.readthedocs.io/en/stable/dev/plugins.html#read-configuration-options) uses the [Confuse](https://confuse.readthedocs.io/en/latest/usage.html#basic-usage) library, it is easy to add `region` configuration value validation during plugin initialization.

Example error message:
```bash
configuration error: audible.region: must be one of ['fr', 'it', 'au', 'us', 'es', 'jp', 'ca', 'de', 'uk', 'in'], not 'jpg'
```

As for the `region` field setting during import:
- The `book_level_region_switch` function will not set an incorrect value. You will simply see that the value remains the same.
```bash
Enter region code (jp, in, de, es, au, fr, uk, us, it, ca) [jp][--]:  usd
Current book region code: --
```
- As I understand the Beets workings, the `region` field will be stored in the database and will not be written to or read from file tags. Instead, the `album_url` field acts as a region's 'proxy value'.

- Finally, the `AlbumInfo` object always gets the `region` value from the Audnex API response.

Reply:
> Another place I could think of where the value might be set incorrectly is in the database, by directly changing it.

That's definitely out of scope and not something we need to worry about, thanks for adding the config validation.